agents/tektronix3021c/tektronix_agent.py

```diff
@@ -0,0 +1,146 @@
+
+
+    def set_frequency(self, session, params=None):
+        """
+        Sets frequency of function generator:
+
+        Args:
+            frequency (float): Frequency to set. Must be between 0 and 25,000,000
+        """
+
+        with self.lock.acquire_timeout(1) as acquired:
+            if acquired:
+                freq = params['frequency']
+                self.awg.setFreq(freq)
+                
+                data = {'timestamp': time.time(),
+                       'block_name': "AWG_frequency",
+                        'data': {'AWG_frequency': freq}
+                       }
+                self.agent.publish_to_feed('AWG', data)
+                
+            else:
+                return False, "Could not acquire lock"
+
+        return True, 'Set frequency {}'.format(params)
+
+    def set_amplitude(self, session, params=None):
+        """
+        Sets current of power supply:
+
+        Args:
+            Amplitude (float): Peak to Peak voltage to set. Must be between 0 and 10.
+        """
+        with self.lock.acquire_timeout(1) as acquired:
+            if acquired:
+                amp = params['amplitude']
+                self.awg.setAmp(amp)
+                
+                data = {'timestamp': time.time(),
+                       'block_name': "AWG_amplitude",
+                        'data': {'AWG_amplitude': amp}
+                       }
+                self.agent.publish_to_feed('AWG', data)
+                
+            else:
+                return False, "Could not acquire lock"
+
+        return True, 'Set amplitude to {} '.format(params)
+
+    def set_output(self, session, params=None):
+        """
+        Task to turn channel on or off.
+
+        Args:
+            state (bool): True for on, False for off.
+        """
+        with self.lock.acquire_timeout(1) as acquired:
+            if acquired:
+                state = params['state']
+                self.awg.setOutput(state)
+                
+                data = {'timestamp': time.time(),
+                       'block_name': "AWG_output",
+                        'data': {'AWG_output': int(state)}
+                       }
+                self.agent.publish_to_feed('AWG', data)
+                
+            else:
+                return False, "Could not acquire lock"
+
+        return True, 'Initialized AWG.'
+
+if __name__ == '__main__':
+    parser = site_config.add_arguments()
+
+    # Add options specific to this agent.
+    pgroup = parser.add_argument_group('Agent Options')
+    pgroup.add_argument('--ip-address')
+    pgroup.add_argument('--gpib-slot')
+
+    # Parse comand line.
+    args = parser.parse_args()
+    # Interpret options in the context of site_config.
+    site_config.reparse_args(args, 'Tektronix AWG')
+
+    agent, runner = ocs_agent.init_site_agent(args)
+
+    p = TektronixAWGAgent(agent, args.ip_address, int(args.gpib_slot))
+
+    agent.register_task('init', p.init_awg)
```

Comment:
This init task could be run at startup if it's convenient with the `startup=True` argument. See the ocs docs for more details: https://ocs.readthedocs.io/en/latest/developer/agents.html#example-agent